src/firewall/core/logger.py

```diff
@@ -134,25 +135,30 @@ def __init__(self, filename, mode="w"):
     def open(self):
         if self.fd:
             return
-        self.fd = open(self.filename, self.mode)
+        flags = os.O_CREAT|os.O_WRONLY
+        if self.mode.startswith('a'):
+            flags |= os.O_APPEND
+        self.fd = os.open(self.filename, flags, 0o640)
+        # Make sure that existing file has correct perms
+        os.chmod(self.fd, 0o640)
```

Comment:
You need to use fchmod() with an fd. chmod()'s first argument is a string/path. This is what is causing all the tests to fail.

Reply:
oops you are right of course. I wonder why I missed it during testing. Thank you